SPHINXsys/src/shared/particle_dynamics/solid_dynamics/contact_dynamics.cpp

```diff
@@ -279,8 +236,10 @@ namespace SPH
 					Real p_star = 0.5 * (p_i + contact_density_k[index_j] * solid_k->ContactStiffness());
```

Comment:
@Xiangyu-Hu if the contact_density is not caluclated for both contacting bodies, contact_density_k[index_j] will be zero here, for all particles. It influences also the contact_force.